components/Chips/src/MDCChipField.m

```diff
@@ -0,0 +1,573 @@
+    chipTextField.autocorrectionType = UITextAutocorrectionTypeNo;
+    chipTextField.autocapitalizationType = UITextAutocapitalizationTypeNone;
+    chipTextField.keyboardType = MDCChipFieldDefaultKeyboardType;
+    [chipTextField addTarget:self
+                      action:@selector(textFieldDidChange)
+            forControlEvents:UIControlEventEditingChanged];
+    [self addSubview:chipTextField];
+    _textField = chipTextField;
+  }
+  return self;
+}
+
+- (void)setBackgroundColor:(UIColor *)backgroundColor {
+  [super setBackgroundColor:backgroundColor];
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+
+  // Calculate the frames for all the chips and set them.
+  NSArray *chipFrames = [self chipFramesForSize:self.bounds.size];
+  for (NSUInteger index = 0; index < _chips.count; index++) {
+    MDCChipView *chip = _chips[index];
+    chip.frame = [chipFrames[index] CGRectValue];
+  }
+
+  // Get the last chip frame and calculate the text field frame from that.
+  CGRect lastChipFrame = [chipFrames.lastObject CGRectValue];
+  self.textField.frame = [self frameForTextFieldForLastChipFrame:lastChipFrame
+                                                   chipFieldSize:self.bounds.size];
+
+  [self updateTextFieldPlaceholderText];
+}
+
+- (void)updateTextFieldPlaceholderText {
+  // Place holder label should be hidden if showPlaceholderWithChips is NO and there are chips.
+  // MDCTextField sets the placeholderLabel opacity to 0 if the text field has no text.
+  self.textField.placeholderLabel.hidden =
+      (!self.showPlaceholderWithChips && self.chips.count > 0) || ![self isTextFieldEmpty];
+}
+
++ (UIFont *)textFieldFont {
+  return [UIFont systemFontOfSize:[UIFont systemFontSize]];
+}
+
+- (CGSize)intrinsicContentSize {
+  CGFloat minWidth =
+      MAX(self.minTextFieldWidth + self.contentEdgeInsets.left + self.contentEdgeInsets.right,
+          self.bounds.size.width);
+  return [self sizeThatFits:CGSizeMake(minWidth, CGFLOAT_MAX)];
+}
+
+- (CGSize)sizeThatFits:(CGSize)size {
+  NSArray *chipFrames = [self chipFramesForSize:size];
+  CGRect lastChipFrame = [chipFrames.lastObject CGRectValue];
+  CGRect textFieldFrame = [self frameForTextFieldForLastChipFrame:lastChipFrame
+                                                    chipFieldSize:size];
+
+  // Calculate the required size off the text field.
+  // To properly apply bottom inset: Calculate what would be the height if there were a chip
+  // instead of the text field. Then add the bottom inset.
+  CGFloat height = CGRectGetMaxY(textFieldFrame) + self.contentEdgeInsets.bottom +
+      (self.chipHeight - textFieldFrame.size.height) / 2;
+  CGFloat width = MAX(size.width, self.minTextFieldWidth);
+
+  return CGSizeMake(width, height);
+}
+
+- (void)clearTextInput {
+  self.textField.text = kMDCEmptyTextString;
+  [self updateTextFieldPlaceholderText];
+}
+
+- (void)setChips:(NSArray<MDCChipView *> *)chips {
```

Comment:
I think a custom getter is needed to return an immutable copy of `_chips` for safety.

Reply:
Done.